controller/scheduler-v2/formation.go

```diff
@@ -51,3 +51,23 @@ func (f *Formation) Update(procs map[string]int) map[string]int {
 	f.Processes = procs
 	return diff
 }
+
+type formationCounts map[utils.FormationKey]map[string]int
+
+func NewFormationCounts(fs Formations) formationCounts {
+	fc := make(formationCounts)
+	for fKey := range fs {
+		fc[fKey] = make(map[string]int)
```

Comment:
Do we need to create all these maps upfront, given they are lazy initialised in `AddJob`?